sql/core/src/main/scala/org/apache/spark/sql/streaming/DataStreamWriter.scala

```diff
@@ -240,8 +241,16 @@ final class DataStreamWriter[T] private[sql](ds: Dataset[T]) {
    * path as new data arrives. The returned [[StreamingQuery]] object can be used to interact with
    * the stream.
    *
+   * @throws TimeoutException If another run of the same streaming query (a streaming query sharing
```

Comment:
> If another run of the same streaming query (a streaming query sharing the same checkpoint location) is already active on the same Spark cluster

"same Spark cluster" seems misleading. AFAIK Spark cannot do it as Spark doesn't have centralized control center - it's just per driver application. Maybe "same Spark application"?

Reply:
nit: `same streaming query (that is, a streaming query`

Reply:
same Spark driver?